[SaferC++] Improve memory safety in InjectedBundle/../WKBundle.cpp

zakariaridouh opened the pull request:
#### 42e27ad9f7ca7437a3e15840695e0372929ffca7
<pre>
[SaferC++] Improve memory safety in InjectedBundle/../WKBundle.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=297886">https://bugs.webkit.org/show_bug.cgi?id=297886</a>
<a href="https://rdar.apple.com/159150342">rdar://159150342</a>

Reviewed by Charlie Wolfe.

Fix SaferC++ issues in `InjectedBundle/API/c/WKBundle.cpp`

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundle.cpp:
(WKBundleSetClient):
(WKBundleSetServiceWorkerProxyCreationCallback):
(WKBundlePostMessage):
(WKBundlePostSynchronousMessage):
(WKBundleGarbageCollectJavaScriptObjects):
(WKBundleGarbageCollectJavaScriptObjectsOnAlternateThreadForDebugging):
(WKBundleGetJavaScriptObjectsCount):
(WKBundleAddOriginAccessAllowListEntry):
(WKBundleRemoveOriginAccessAllowListEntry):
(WKBundleResetOriginAccessAllowLists):
(WKBundleSetAsynchronousSpellCheckingEnabledForTesting):
(WKBundleGetLiveDocumentURLsForTesting):
(WKBundleCreateWKDataFromUInt8Array):
(WKBundleNumberOfPages):
(WKBundlePageNumberForElementById):
(WKBundlePageSizeAndMarginsInPixels):
(WKBundleIsPageBoxVisible):
(WKBundleSetUserStyleSheetLocationForTesting):
(WKBundleRemoveAllWebNotificationPermissions):
(WKBundleCopyWebNotificationID):
(WKBundleSetTabKeyCyclesThroughElements):
(WKBundleExtendClassesForParameterCoder):

Canonical link: <a href="https://commits.webkit.org/299516@main">https://commits.webkit.org/299516@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a3b9cacd839fdcad2fd46c06e15ab0dab9cfe718

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119147 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38828 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29479 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125366 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71210 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/20c24d16-d632-498d-a622-540d15de02a7) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39524 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47410 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90495 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59884 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a68faa28-522d-46f9-ab91-37a4f28b9e79) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122100 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31504 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106799 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70918 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fff1b58f-b6ae-49e2-a410-2018df797ea8) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30553 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69017 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100950 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25097 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128388 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46054 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34796 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99067 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46421 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103015 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98847 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25155 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44316 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22311 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42633 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45924 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51604 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45391 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48737 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47079 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a3b9cacd839fdcad2fd46c06e15ab0dab9cfe718. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49868

zakariaridouh committed:
[SaferC++] Improve memory safety in InjectedBundle/../WKBundle.cpp
https://bugs.webkit.org/show_bug.cgi?id=297886
rdar://159150342

Reviewed by Charlie Wolfe.

Fix SaferC++ issues in `InjectedBundle/API/c/WKBundle.cpp`

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundle.cpp:
(WKBundleSetClient):
(WKBundleSetServiceWorkerProxyCreationCallback):
(WKBundlePostMessage):
(WKBundlePostSynchronousMessage):
(WKBundleGarbageCollectJavaScriptObjects):
(WKBundleGarbageCollectJavaScriptObjectsOnAlternateThreadForDebugging):
(WKBundleGetJavaScriptObjectsCount):
(WKBundleAddOriginAccessAllowListEntry):
(WKBundleRemoveOriginAccessAllowListEntry):
(WKBundleResetOriginAccessAllowLists):
(WKBundleSetAsynchronousSpellCheckingEnabledForTesting):
(WKBundleGetLiveDocumentURLsForTesting):
(WKBundleCreateWKDataFromUInt8Array):
(WKBundleNumberOfPages):
(WKBundlePageNumberForElementById):
(WKBundlePageSizeAndMarginsInPixels):
(WKBundleIsPageBoxVisible):
(WKBundleSetUserStyleSheetLocationForTesting):
(WKBundleRemoveAllWebNotificationPermissions):
(WKBundleCopyWebNotificationID):
(WKBundleSetTabKeyCyclesThroughElements):
(WKBundleExtendClassesForParameterCoder):

Canonical link: https://commits.webkit.org/299516@main
